MdeModulePkg/Core/Dxe/Image/Image.c

```diff
@@ -753,7 +766,36 @@ CoreLoadPeImage (
   if (EFI_ERROR (Status)) {
     goto Done;
   }
+  // MU_CHANGE START
+  if (mMemoryAttribute == NULL) {
+    gBS->LocateProtocol (
+            &gEfiMemoryAttributeProtocolGuid,
+            NULL,
+            (VOID **) &mMemoryAttribute
+            );
+  }
 
+  if (mMemoryAttribute != NULL) {
+
+    Status = mMemoryAttribute->GetMemoryAttributes (
+                                mMemoryAttribute,
+                                Image->ImageContext.ImageAddress,
+                                EFI_PAGES_TO_SIZE (Image->NumberOfPages),

```

Comment:
Out of curiosity, does this region cover only the code? Or both the code and data?

Reply:
Both code and data. Later, when ProtectUefiImageMu() is called, permissions are updated appropriately.